### PR TITLE
Handle error when file not found

### PR DESCRIPTION
**Pull request by kevinlitchfield**

### CODE REVIEW

File: icdiff [hunk 1]
```diff
@@ -535,8 +535,12 @@ def diff(options, a, b):
...
 
     if is_a_file and is_b_file:
-        if not filecmp.cmp(a, b, shallow=False):
-            diff_files(options, a, b)
+        try:
+            if not filecmp.cmp(a, b, shallow=False):
+                diff_files(options, a, b)
+        except OSError as e:
+            if e.errno == errno.ENOENT:
+                print_meta("error: file '%s' was not found" % e.filename)
```
Comment (jeffkaufman) [inline icdiff]: This will swallow any `OSError` exceptions where the `errno` isn't `ENOENT`.  We should either print or raise in response to any errors.

Comment (kevinlitchfield) [inline icdiff]: Good point, thanks. Have fixed.

**Comment by kevinlitchfield** [comment]
> In what situations do we run into errors from `filecmp.cmp` Why would we be here if a file didn't exist? Is this race conditions with deleting files or something else?

By the time we reach `filecmp.cmp`, we haven't checked that the files exist. We've only checked `os.path.isdir`, which just returns `False` when passed a nonexistent path. So whenever a file doesn't exist, `filecmp.cmp` will error.

**Comment by jeffkaufman** [comment]
If you fix the hiding errors thing I commented on then I'm good to merge

**Comment by jeffkaufman** [comment]
f07d683 looks good

**Pull request merged**